test/functorch/test_ops.py

```diff
@@ -827,6 +830,7 @@ def test_vmapvjp(self, device, dtype, op):
         # ---------------------------- BUGS ------------------------------------
         # The following are bugs that we should fix
         decorate('nn.functional.conv2d', decorator=unittest.skipIf(IS_ARM64, "Fails on M1")),
+        decorate('linalg.det', 'singular', decorator=unittest.skipIf(IS_MACOS, "Fails on x86 MacOS CI")),
```

Comment:
Why did this just show up now?
Also, is this a silent correctness problem or something else?

Reply:
I am not really sure. It only happens on Mac with x86-64 and not on ARM (based on CI failure and trying to reproduce on M1 Mac - https://github.com/pytorch/pytorch/pull/86202#issuecomment-1272996317)

Reply:
Do you have the logs from a failure?

Reply:
Sorry, I thought I had added the CI failure in the [comment](https://github.com/pytorch/pytorch/pull/86202#issuecomment-1272996317). Have updated it now.

Also, here is the failure snippet for reference:
```
=================================== FAILURES ===================================
_______ TestOperatorsCPU.test_vmapjvpall_linalg_det_singular_cpu_float32 _______

self = <test_ops.TestOperatorsCPU testMethod=test_vmapjvpall_linalg_det_singular_cpu_float32>
device = 'cpu', dtype = torch.float32
op = OpInfo(name='linalg.det', ref=None, aliases=(<torch.testing._internal.opinfo.core.AliasInfo object at 0x7fc3f01ce6d0>,...s=True, test_neg_view=True, assert_jit_shape_analysis=False, supports_expanded_weight=False, is_factory_function=False)

    @ops(op_db + additional_op_db, allowed_dtypes=(torch.float,))
    @opsToleranceOverride('TestOperators', 'test_vmapjvpall', (
        tol1('nn.functional.conv_transpose3d',
             {torch.float32: tol(atol=2e-04, rtol=9e-3)}, device_type='cuda'),
        tol1('linalg.householder_product',
             {torch.float32: tol(atol=2e-04, rtol=9e-3)}, device_type='cuda'),
        tol1('linalg.householder_product',
             {torch.float32: tol(atol=2e-04, rtol=1e-4)}, device_type='cpu'),
    ))
    @skipOps('TestOperators', 'test_vmapjvpall', vmapjvpall_fail)
    @toleranceOverride({torch.float32: tol(atol=1e-04, rtol=1e-04)})
    # This is technically a superset of test_vmapjvp. We should either delete test_vmapjvp
    # or figure out if we can split vmapjvpall. It's useful to keep test_vmapjvp intact
    # because that coresponds to "batched forward-mode AD" testing in PyTorch core
    def test_vmapjvpall(self, device, dtype, op):
        if is_inplace(op, op.get_op()):
            # TODO: test in-place
            self.skipTest("Skipped! NYI: inplace-testing not supported.")
            return
    
        samples = op.sample_inputs(device, dtype, requires_grad=False)
    
        if not op.supports_forward_ad:
            self.skipTest("Skipped! Forward AD not supported.")
            return
    
        for sample in samples:
            arg_values = [sample.input] + list(sample.args)
            kwarg_values = sample.kwargs
            args = tuple(arg_values) + tuple(kwarg_values)
            fn, args = get_jvp_variant_primals_tangents(op, sample)
            is_batch_norm_and_training = is_batch_norm_training(op.name, kwarg_values)
            generator = get_fallback_and_vmap_exhaustive(
                fn, args, {}, is_batch_norm_and_training=is_batch_norm_and_training)
            for vmap_expected, vmap_actual, vmapvmap_expected, vmapvmap_actual in generator:
>               self.assertEqual(vmap_expected, vmap_actual)
E               AssertionError: Tensor-likes are not close!
E               
E               Mismatched elements: 2 / 8 (25.0%)
E               Greatest absolute difference: 0.0353260412812233 at index (0, 0, 0) (up to 0.0001 allowed)
E               Greatest relative difference: inf at index (0, 0, 0) (up to 0.0001 allowed)
E               
E               The failure occurred for item [1]

functorch/test_ops.py:884: AssertionError
=============================== warnings summary ===============================
```

Reply:
NB: I'd merge this PR first and then continue investigating later (so you don't need to deal with rebasing or merge conflicts)